### PR TITLE
Add new tests to increase coverage and some clean-up

### PR DESCRIPTION
**Pull request by lbenet**
Among the things that I have cleaned up, I commented some convert
methods which I think speeded-down any calculation. I also
commented the function `variables` which was never used
(utils_TaylorN.jl).

### CODE REVIEW

**Comment by lbenet** [comment]
I think this is ready to be merged.

@dpsanders Do you know how to test `show` methods?

**Comment by dpsanders** [comment]
Here's one possible way to get the output of `show` methods as a string:
```julia
julia> buffer = IOBuffer()
IOBuffer(data=Uint8[...], readable=true, writable=true, seekable=true, append=false, size=0, maxsize=Inf, ptr=1, mark=-1)

julia> show(buffer, 3)

julia> bytestring(buffer)
"3"
```
Now that I think of it, can't you just do something like the following, using `string`?
```julia
julia> using TaylorSeries
x, y =
julia> x, y = set_variables("x y")
2-element Array{TaylorN{Float64},1}:
  1.0 x + 𝒪(‖x‖⁷)
  1.0 y + 𝒪(‖x‖⁷)

julia> x
 1.0 x + 𝒪(‖x‖⁷)

julia> string(x)
" 1.0 x + 𝒪(‖x‖⁷)"
```

**Comment by dpsanders** [comment]
Apparently `string` does in fact use `IOBuffer` internally. I think instead of `bytestring` the function should be `takebuf_string` (a truly awful name for a function!)

File: src/utils_Taylor1.jl [hunk 1]
```diff
@@ -72,8 +72,8 @@ end
 convert{T<:Number, S<:Number}(::Type{Taylor1{T}}, b::Array{S,1}) =
     Taylor1(convert(Array{T,1},b))
 convert{T<:Number}(::Type{Taylor1{T}}, b::Number) = Taylor1([convert(T,b)], 0)
-convert{T<:Number}(::Type{Taylor1{T}}, a::Taylor1{T}) = a
-convert{T<:Number}(::Type{Taylor1{T}}, b::Array{T,1}) = Taylor1(b)
+# convert{T<:Number}(::Type{Taylor1{T}}, a::Taylor1{T}) = a
```
Comment (dpsanders) [inline src/utils_Taylor1.jl]: I would suggest just deleting all these methods, instead of commenting them out.

**Comment by lbenet** [comment]
I've followed your first suggestion, using `IOBuffer()` and `bytestring`, which I find is quite honest.

I think there is still something to be done to get a better coverage, since some functions in TaylorSeries.jl and hashtables.jl still have functions that remain untested.

In any case, tests are passing.

**Comment by lbenet** [comment]
My mistake. I'm leaving tomorrow so I wanted to get a couple of things done before...

I indeed agree that your second suggestion is much more readable, now that I understand it. I can submit a new commit to this branch, but this will be during this afternoon.

Incidentally, do you know why coveralls hasn't updated the bagde?

**Comment by dpsanders** [comment]
Don't worry, I already did it in #20 ;)

**Pull request merged**